File: backend/tildau/src/main/java/kz/kbtu/tildau/dto/exercise/SubmitExerciseResponse.java
package kz.kbtu.tildau.dto.exercise;

import kz.kbtu.tildau.dto.nextStep.NextStepResponse;
import kz.kbtu.tildau.dto.stats.AchievementResponse;
import lombok.*;

import java.util.List;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Builder
public class SubmitExerciseResponse {
    private UUID attemptId;
    private int overallScore;
    private List<String> feedback;
    private NextStepResponse nextStep;
    private List<AchievementResponse> newAchievements;
}

File: android/Tildau2/app/src/main/java/com/example/tildau/data/model/exercise/SubmitExerciseResponse.kt
package com.example.tildau.data.model.exercise

import com.example.tildau.data.model.next.NextStepResponse
import com.example.tildau.data.model.stats.AchievementResponse

data class SubmitExerciseResponse(
    val attemptId: String?,
    val overallScore: Int,
    val feedback: List<String>,
    val nextStep: NextStepResponse,
    val newAchievements: List<AchievementResponse> = emptyList()
)